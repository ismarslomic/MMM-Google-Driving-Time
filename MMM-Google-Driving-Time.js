/*! *****************************************************************************
  mmm-google-driving-time
  Version 1.0.0

  Get info from Google when to start driving based on shortest duration, for the MagicMirrorÂ² platform.
  Please submit bugs at https://github.com/ismarslomic/MMM-Google-Driving-Time/issues

  (c) ismar@slomic.no
  Licence: MIT

  This file is auto-generated. Do not edit.
***************************************************************************** */

!function(t){"use strict";function e(t){var e=Object.create(null);return t&&Object.keys(t).forEach((function(i){if("default"!==i){var n=Object.getOwnPropertyDescriptor(t,i);Object.defineProperty(e,i,n.get?n:{enumerable:!0,get:function(){return t[i]}})}})),e.default=t,Object.freeze(e)}var i,n,a=e(t);!function(t){t.DRIVING_TIME_REQUEST="DRIVING_TIME_REQUEST",t.DRIVING_TIME_SUCCESS_RESPONSE="DRIVING_TIME_SUCCESS_RESPONSE",t.DRIVING_TIME_FAILED_RESPONSE="DRIVING_TIME_FAILED_RESPONSE"}(i||(i={})),function(t){t.TODAY="TODAY",t.TOMORROW="TOMORROW"}(n||(n={})),Module.register("MMM-Google-Driving-Time",{requiresVersion:"2.19.0",defaults:{lastUpdateFormat:"HH:mm",updateIntervalInSeconds:5,origin:void 0,destination:void 0,language:void 0,trafficModel:"best_guess",apiKey:void 0,departureDate:n.TODAY,departureTimes:[{hours:13,minutes:30}]},getScripts:()=>["moment.js"],getStyles:()=>["MMM-Google-Driving-Time.css"],getTranslations:()=>({en:"translations/en.json",de:"translations/de.json"}),getTemplate:()=>"templates/MMM-Google-Driving-Time.njk",getTemplateData(){var t,e;return{config:this.config,lastUpdate:moment(null===(t=this.state)||void 0===t?void 0:t.lastUpdate).format(this.config.lastUpdateFormat),drivingTime:null===(e=this.state)||void 0===e?void 0:e.drivingTime}},loading:!0,start(){this.loadData(),this.scheduleUpdate(),this.updateDom()},scheduleUpdate(){this.config.updateIntervalInSeconds=this.config.updateIntervalInSeconds<120?120:this.config.updateIntervalInSeconds,setInterval((()=>{this.loadData()}),1e3*this.config.updateIntervalInSeconds)},loadData(){this.sendSocketNotification(i.DRIVING_TIME_REQUEST,this.config)},socketNotificationReceived(t,e){t===i.DRIVING_TIME_SUCCESS_RESPONSE&&(this.state=e,a.log(i.DRIVING_TIME_SUCCESS_RESPONSE,this.state),this.updateDom())}})}(Log);
