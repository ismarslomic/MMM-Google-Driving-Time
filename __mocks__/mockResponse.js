const generateResponse = () => ({
  status: 200,
  statusText: 'OK',
  headers: {},
  config: {},
  request: {},
  data: {
    'destination_addresses': [
      'GÃ¸teborggata 32, 0566 Oslo, Norge'
    ],
    'origin_addresses': [
      'Hemsilbreidda 16, 3560 Hemsedal, Norge'
    ],
    'rows': [
      {
        'elements': [
          {
            'distance': {
              'text': '203 km',
              'value': 203466
            },
            'duration': {
              'text': '3 timer 0 min',
              'value': 10782
            },
            'duration_in_traffic': {
              'text': '2 timer 48 min',
              'value': 10064
            },
            'status': 'OK'
          }
        ]
      }
    ],
    'status': 'OK'
  }
})

module.exports = { generateResponse }

