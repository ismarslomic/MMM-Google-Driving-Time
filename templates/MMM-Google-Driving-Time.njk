<div class="dimmed light small">
  <div class="jast-stock">To: {{ drivingTime.destination }} ( {{ drivingTime.distance.description }} )</div>
  <div class="jast-stock">Duration: {{ drivingTime.durationDiff.description }}</div>
  <div class="jast-stock">{{ "lastUpdate" | translate | safe }} {{ lastUpdate }}</div>
</div>
