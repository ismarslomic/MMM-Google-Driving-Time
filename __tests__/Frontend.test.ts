describe('Frontend', () => {
  beforeAll(() => {
    require('../__mocks__/Logger')
    require('../__mocks__/Module')
  })

  const name = 'MMM-Google-Driving-Time'

  let MMMGoogleDriveTime

  beforeEach(() => {
    jest.resetModules()
    require('../MMM-Google-Driving-Time')

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    MMMGoogleDriveTime = global.Module.create(name)
    MMMGoogleDriveTime.setData({ name, identifier: `Module_1_${name}` })
  })

  test('requires version 2.19', () => {
    expect(MMMGoogleDriveTime.requiresVersion).toBe('2.19.0')
  })

  test('has correct defaults', () => {
    expect(MMMGoogleDriveTime.defaults).toMatchSnapshot()
  })

  test('inits module with state loading === true', () => {
    expect(MMMGoogleDriveTime.loading).toBe(true)
  })
})
