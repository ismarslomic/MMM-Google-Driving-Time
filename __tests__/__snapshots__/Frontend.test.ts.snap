// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontend has correct defaults 1`] = `
Object {
  "apiKey": undefined,
  "departureDate": "TODAY",
  "departureTimes": Array [
    Object {
      "hours": 13,
      "minutes": 30,
    },
  ],
  "destination": undefined,
  "language": undefined,
  "lastUpdateFormat": "HH:mm",
  "origin": undefined,
  "trafficModel": "best_guess",
  "updateIntervalInSeconds": 5,
}
`;
